waspc/src/Wasp/Generator/DbGenerator/Operations.hs

```diff
@@ -60,3 +66,40 @@ copyMigrationsBackToSource genProjectRootDirAbs dbMigrationsDirInWaspProjectDirA
   where
     waspMigrationsDir = SP.castDir dbMigrationsDirInWaspProjectDirAbs
     genProjectMigrationsDir = SP.castDir $ genProjectRootDirAbs SP.</> dbRootDirInProjectRootDir SP.</> dbMigrationsDirInDbRootDir
+
+writeDbSchemaChecksumToFile :: Path' Abs (Dir ProjectRootDir) -> Path' (Rel ProjectRootDir) File' -> IO ()
+writeDbSchemaChecksumToFile genProjectRootDirAbs dbSchemaChecksumInProjectRootDir = do
+  dbSchemaExists <- doesFileExist dbSchemaFp
+  when dbSchemaExists $ do
+    checksum <- hexToString <$> checksumFromFilePath dbSchemaFp
+    writeFile dbSchemaChecksumFp checksum
+  where
+    dbSchemaFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaFileInProjectRootDir
+    dbSchemaChecksumFp = SP.fromAbsFile $ genProjectRootDirAbs SP.</> dbSchemaChecksumInProjectRootDir
+
+generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> IO (Either String ())
+generatePrismaClient genProjectRootDirAbs = do
+  dbSchemaChecksumFileExists <- doesFileExist dbSchemaChecksumFp
+
+  if dbSchemaChecksumFileExists
+    then do
+      dbSchemaFileChecksum <- hexToString <$> checksumFromFilePath dbSchemaFp
+      dbChecksumFileContents <- readFile dbSchemaChecksumFp
+      if dbSchemaFileChecksum == dbChecksumFileContents then return $ Right () else generatePrismaClient'
+    else generatePrismaClient'
```

Comment:
To make this piece of code a bit easier to read, I would consider rewriting it to
```hs
ifM wasCurrentSchemaAlreadyGenerated (return $ Right()) generatePrismaClient'

...


where
  wasCurrentSchemaAlreadyGenerated :: IO Bool
  wasCurrentSchemaAlreadyGenerated = ...
```

Reply:
btw I never before used `ifM` but I just thought "there must be some way to do if-then-else on monadic value and hoogle said there is ifM!

Reply:
Given that this was an external library, and to not overly populate our Util.hs file, I decided to punt on this one for now. Additionally, when trying to make use of it to clean things up I felt it did not make it that much easier to follow (for me at least) with the fact that we have a file existence check before we can check the contents. That outer check either had to wrap the `ifM` (and move the read farther away from the FilePath check, which felt off) or build it into the helper, but then we didn't clean up the shape too much in the latter case. I appreciate the idea though and sorry if I misunderstood 👍🏻